dashboard/modules/snapshot/tests/test_job_submission.py

```diff
@@ -35,63 +36,120 @@ def _get_snapshot(address: str):
 def test_successful_job_status(
     ray_start_with_dashboard, disable_aiohttp_cache, enable_test_module
 ):
-    address = ray_start_with_dashboard["webui_url"]
+    address = ray_start_with_dashboard.address_info["webui_url"]
     assert wait_until_server_available(address)
     address = format_web_url(address)
 
+    job_sleep_time_s = 5
     entrypoint_cmd = (
-        'python -c"' "import ray;" "ray.init();" "import time;" "time.sleep(5);" '"'
+        'python -c"'
+        "import ray;"
+        "ray.init();"
+        "import time;"
+        f"time.sleep({job_sleep_time_s});"
+        '"'
     )
 
     client = JobSubmissionClient(address)
-    job_id = client.submit_job(entrypoint=entrypoint_cmd)
+    start_time_s = int(time.time())
+    runtime_env = {"env_vars": {"RAY_TEST_123": "123"}}
+    metadata = {"ray_test_456": "456"}
+    job_id = client.submit_job(
+        entrypoint=entrypoint_cmd, metadata=metadata, runtime_env=runtime_env
+    )
 
     def wait_for_job_to_succeed():
         data = _get_snapshot(address)
+        legacy_job_succeeded = False
+        job_succeeded = False
+
+        # Test legacy job snapshot (one driver per job).
         for job_entry in data["data"]["snapshot"]["jobs"].values():
             if job_entry["status"] is not None:
                 assert job_entry["config"]["metadata"]["jobSubmissionId"] == job_id
                 assert job_entry["status"] in {"PENDING", "RUNNING", "SUCCEEDED"}
                 assert job_entry["statusMessage"] is not None
-                return job_entry["status"] == "SUCCEEDED"
-
-        return False
+                legacy_job_succeeded = job_entry["status"] == "SUCCEEDED"
+
+        # Test new jobs snapshot (0 to N drivers per job).
+        for job_submission_id, entry in data["data"]["snapshot"][
+            "jobSubmission"
+        ].items():
+            if entry["status"] is not None:
+                assert entry["status"] in {"PENDING", "RUNNING", "SUCCEEDED"}
+                assert entry["statusMessage"] is not None
+                # TODO(architkulkarni): Disable automatic camelcase.
+                assert entry["runtimeEnv"] == {"envVars": {"RAYTest123": "123"}}
```

Comment:
This is bad, we don't want to convert arbitrary strings in the job data to camelcase.  It turns out we've had this problem on the existing jobs snapshot the whole time:

```
'03000000': {'config': {'metadata': {'jobName': 'raysubmit_RcjHs96ykN3SQDB1',
                                                                     'jobSubmissionId': 'raysubmit_RcjHs96ykN3SQDB1',
                                                                     'rayTest789': '789'},
                                                        'namespace': 'fa5d38e3-a819-488d-921e-a83b62a5d873',
                                                        'runtimeEnv': {'envVars': {'RAYTest456': '456'}}},
```
The env vars are incorrect.  (The camelcase algorithm seems nontrivial; how does `RAY_TEST_456` get converted to `RAYTest456`??)

Reply:
Previous discussion on disabling the camel casing: https://github.com/ray-project/ray/pull/17977 

It seems like everyone is okay with it, but we're worried about breaking backwards compatibility.

Reply:
I'd propose we don't block this PR on this issue, and instead make a separate issue to disable camelcasing in the snapshot altogether.

Reply:
@architkulkarni can you make sure to follow up on this? it's sure to come back to bite us...